core/clients/metadataclients/metadata-clients.go

```diff
@@ -301,6 +302,46 @@ func (d *DeviceRestClient) decodeDevice(resp *http.Response) (models.Device, err
 	return dev, err
 }
 
+//Use the models.Event.Device property for the supplied token parameter.
+//The above property is currently double-purposed and needs to be refactored.
+//This call replaces the previous two calls necessary to lookup a device by id followed by name.
+func (d *DeviceRestClient) CheckForDevice(token string) (models.Device, error) {
+	req, err := http.NewRequest(http.MethodGet, d.url+"/check/"+token, nil)
+	if err != nil {
+		fmt.Println(err)
```

Comment:
with this https://github.com/edgexfoundry/edgex-go/pull/177 fmt is going to be replaced with logging-client. Also you already use this in code below. So could we have same approach everywhere?

Reply:
I have no intention of accepting PR 177 because I'm not in agreement with it. See this thread
https://github.com/edgexfoundry/edgex-go/issues/131

Therefore I'm not taking those code changes into account. If someone else accepts the PR, so be it.